src/couch_mrview/src/couch_mrview_util.erl

```diff
@@ -1070,16 +1077,33 @@ old_view_format(View, SI, KSI) ->
         View#mrview.options
     }.
 
-maybe_update_header(#mrheader{} = Header) ->
-    Header;
-maybe_update_header(Header) when tuple_size(Header) == 6 ->
+maybe_update_header(#mrheader{view_info = Info} = Header) when is_map(Info) ->
+    % Latest (3.2.1+) version. Note: the size of the record is the same as the
+    % legacy <2.3.1 version. The main difference is that the LogBt field is now
+    % a map. This trick also allows for easy downgrading back to version 3.2.1
+    % and then upgrading back to 3.2.1 if needed.
```

Comment:
`back to 3.2.1+ if needed.`?

Reply:
Good catch, yeah it would be `back to 3.2.1+`. So it would allow downgrading, then upgrading again without resetting views as in the 3.0.0-3.2.1 versions that member of the header is ignored.